operators/gstreamer/CMakeLists.txt

```diff
@@ -0,0 +1,142 @@
+
+# Find GStreamer
+pkg_check_modules(GSTREAMER REQUIRED gstreamer-1.0)
+pkg_check_modules(GSTREAMER_BASE REQUIRED gstreamer-base-1.0)
+pkg_check_modules(GSTREAMER_VIDEO REQUIRED gstreamer-video-1.0)
+pkg_check_modules(GSTREAMER_CUDA gstreamer-cuda-1.0)  # Optional - only available in GStreamer 1.24+
+pkg_check_modules(GSTREAMER_APP REQUIRED gstreamer-app-1.0)
+
+# Create the Holoscan GStreamer Bridge library
+add_library(holoscan_gstreamer_bridge SHARED
+  gst/buffer.cpp
+  gst/caps.cpp
+  gst/memory.cpp
+  gst/message.cpp
+  gst/video_info.cpp
+  gst_src_bridge.cpp
+  gst_video_recorder_operator.cpp
+)
+
+# Create alias for consistent naming
+add_library(holoscan::ops::gstreamer_bridge ALIAS holoscan_gstreamer_bridge)
+
+# Set target properties
+set_target_properties(holoscan_gstreamer_bridge PROPERTIES
+  C_STANDARD 99
+  C_STANDARD_REQUIRED ON
+  CXX_STANDARD 17
+  CXX_STANDARD_REQUIRED ON
+  POSITION_INDEPENDENT_CODE ON
+)
+
+# Set include directories (PUBLIC so they propagate to applications)
+target_include_directories(holoscan_gstreamer_bridge
+    PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+    $<INSTALL_INTERFACE:include>
+    ${GSTREAMER_INCLUDE_DIRS}
+    ${GSTREAMER_BASE_INCLUDE_DIRS}
+    ${GSTREAMER_VIDEO_INCLUDE_DIRS}
+    ${GSTREAMER_APP_INCLUDE_DIRS}
+)
+
+# Link libraries for custom sink
+target_link_libraries(holoscan_gstreamer_bridge
+    PUBLIC
+    ${GSTREAMER_LIBRARIES}
+    ${GSTREAMER_BASE_LIBRARIES}
+    ${GSTREAMER_VIDEO_LIBRARIES}
+    ${GSTREAMER_APP_LIBRARIES}
+    holoscan::core)
+
+# Set compile flags (PRIVATE since they're compiler-specific)
+target_compile_options(holoscan_gstreamer_bridge PRIVATE
+    ${GSTREAMER_CFLAGS_OTHER}
+    ${GSTREAMER_BASE_CFLAGS_OTHER}
+    ${GSTREAMER_VIDEO_CFLAGS_OTHER}
+    ${GSTREAMER_APP_CFLAGS_OTHER})
+
+# Configure CUDA support (optional - requires GStreamer 1.24+)
+if(GSTREAMER_CUDA_FOUND)
+    message(STATUS "GStreamer CUDA support: ENABLED (found gstreamer-cuda-1.0)")
+
+    # Add CUDA include directories
+    target_include_directories(holoscan_gstreamer_bridge
+        PUBLIC
+        ${GSTREAMER_CUDA_INCLUDE_DIRS}
+    )
+
+    # Add CUDA libraries
+    target_link_libraries(holoscan_gstreamer_bridge
+        PUBLIC
+        ${GSTREAMER_CUDA_LIBRARIES}
+    )
+
+    # Add CUDA compile flags
+    target_compile_options(holoscan_gstreamer_bridge PRIVATE
+        ${GSTREAMER_CUDA_CFLAGS_OTHER}
+    )
+else()
+    message(STATUS "GStreamer CUDA support: DISABLED (gstreamer-cuda-1.0 not found - requires GStreamer 1.24+)")
+endif()
+
+# Platform-specific settings
+if(APPLE)
+    # For macOS, we might need additional frameworks
+    find_library(FOUNDATION_FRAMEWORK Foundation)
+    find_library(COCOA_FRAMEWORK Cocoa)
+    if(FOUNDATION_FRAMEWORK AND COCOA_FRAMEWORK)
+        target_link_libraries(holoscan_gstreamer_bridge ${FOUNDATION_FRAMEWORK} ${COCOA_FRAMEWORK})
+    endif()
+endif()
```

Comment:
We don't currently support macOS in Holoscan SDK, is there a reason to target macOS in this sample app? Suggest removing

```suggestion
```